src/main/java/core/basesyntax/service/strategy/impl/PurchaseOperationService.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.service.strategy.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.strategy.OperationService;
+import core.basesyntax.storage.TemporaryStorage;
+
+public class PurchaseOperationService implements OperationService {
+    @Override
+    public void calculateByOperation(FruitTransaction fruitTransaction) {
+        if (isOnBalanceSheet(fruitTransaction)) {
```

Comment:
`isOnBalance` check is not enough for purchasing operation. What if user try to purchase more quantity that present in storage?